azkaban-common/src/main/java/azkaban/project/JdbcDependencyManager.java

```diff
@@ -91,8 +91,13 @@
       while (rs.next()) {
         // Columns are (starting at index 1): file_name, file_sha1, validation_status
         Dependency d = hashAndFileNameToDep.remove(rs.getString(1) + rs.getString(2));
-        FileValidationStatus v = FileValidationStatus.valueOf(rs.getInt(3));
-        depValidationStatuses.put(d, v);
+
+        // HashMap.remove will return null if the key is not found, hence check for it before
+        // adding to depValidationStatuses.
+        if (d != null) {
```

Comment:
It will be better to add contains check for key in hashAndFileNameToDep instead of null check over here. All of these should be inside that contains if case.

Reply:
Contains check will lead to 2 lookups in the hasMap for every dependency instead of 1 which is the current case. Let me know if you feel strongly about using contains, I can make the change.

Reply:
I agree with @li-gsalia about it.

Reply:
If we are expecting more key miss i.e. keys not present in hashAndFileNameToDep, then contains check looks more logical. But I also agree with @li-gsalia. So will leave it up to @li-gsalia to take a call on this.

Reply:
+1 with @abhisheknath2011

Reply:
There should be no miss based on the current code. Added the null check since set.remove() can return null. This will future proof the code in case future code changes lead to a miss in set.